### ISSUE TITLE
Bundle Size increased Significantly

### ISSUE THREAD

**Issue by mohitarora777** [react/metro#983]
### New Version

0.71.7

### Old Version

0.66.3

### Build Target(s)

Android

### Output of `react-native info`

System:
    OS: macOS 13.3.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 2.28 GB / 32.00 GB
    Shell: 3.2.57 - /bin/sh
  Binaries:
    Node: 20.1.0 - /opt/homebrew/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 9.6.4 - /opt/homebrew/bin/npm
    Watchman: 2023.05.08.00 - /opt/homebrew/bin/watchman
  Managers:
    CocoaPods: 1.11.3 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
    Android SDK: Not Found
  IDEs:
    Android Studio: 2021.3 AI-213.7172.25.2113.9123335
    Xcode: 14.2/14C18 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.17 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.2.0 => 18.2.0 
    react-native: 0.71.7 => 0.71.7 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

### Issue and Reproduction Steps

Upgrade to 0.71 and create a release app bundle, now run apk analyser.
Old=>
![image](https://github.com/facebook/react-native/assets/107910744/d6682277-2688-4365-8fa6-e5c191ed5eb4)

New=>
![image (1)](https://github.com/facebook/react-native/assets/107910744/24f1d018-dc7a-4f86-9d84-a6d067d548c1)

Only 3 dependencies changed in package.json. Rest were changed in devDependencies.
<img width="1507" alt="Screenshot 2023-05-12 at 4 56 26 PM" src="https://github.com/facebook/react-native/assets/107910744/b578ed18-eabc-41f5-a269-ad135fc9f0fb">

**Comment by cortinico** [comment]
Passing this over to the Metro team

**Comment by mohitarora777** [comment]
@cortinico Is Metro team going to see this issue if this is happening because of hermes ?

**Comment by mohitarora777** [comment]
@robhogan can u help here ?

**Comment by saif-o99** [comment]
no updates?

**Comment by mohitarora777** [comment]
@saif-o99 u r facing too

**Comment by saif-o99** [comment]
Yes, my app release apk increased ~10MB after updating RN to latest

**Comment by robhogan** [comment]
Apologies this never got an answer.

This isn't a Metro issue. With Hermes enabled the bundle asset in your APK is Hermes bytecode rather than JavaScript, and bytecode is larger on disk than the equivalent minified JS. That's the difference you're measuring, and it's consistent with the 710KB you saw with Hermes off.

The trade-off is deliberate - bytecode costs disk space but removes parsing and compilation at startup. Metro's output is the JavaScript that goes into `hermesc`, so questions about bytecode or APK size belong at https://github.com/react/react-native.

Closing on that basis.
